pkgs/applications/window-managers/stalonetray/default.nix

```diff
@@ -1,29 +1,63 @@
-{ lib, stdenv, fetchurl, libX11, xorgproto }:
+{ autoconf
+, automake
+, docbook_xml_dtd_44
+, docbook-xsl-ns
+, fetchFromGitHub
+, fetchurl
+, lib
+, libxslt
+, libX11
+, libXpm
+, stdenv
+, xorgproto
+}:
 
 stdenv.mkDerivation rec {
   pname = "stalonetray";
-  version = "0.8.3";
+  version = "0.8.4";
 
-  src = fetchurl {
-    url = "mirror://sourceforge/stalonetray/${pname}-${version}.tar.bz2";
-    sha256 = "0k7xnpdb6dvx25d67v0crlr32cdnzykdsi9j889njiididc8lm1n";
+  src = fetchFromGitHub {
+    owner = "kolbusa";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-grxPqSYPLUstLIOKqzMActaSQ2ftYrjbalfR4HcPDRY=";
   };
 
-  buildInputs = [ libX11 xorgproto ];
+  nativeBuildInputs = [
+    autoconf
+    automake
+    docbook_xml_dtd_44
+    docbook-xsl-ns
+    libxslt
+    libX11
+    libXpm
+  ];
+
+  preConfigure =
+    let
+      db_root = "${docbook-xsl-ns}/share/xml/docbook-xsl-ns";
+      ac_str = "AC_SUBST(DOCBOOK_ROOT)";
+      ac_str_sub = "DOCBOOK_ROOT=${db_root}; ${ac_str}";
+      dtd_str = "http://www.oasis-open.org/docbook/xml/4.4/docbookx.dtd";
+      dtd_str_sub = "${docbook_xml_dtd_44}/xml/dtd/docbook/docbookx.dtd";
+    in
+      ''
+        substituteInPlace configure.ac --replace '${ac_str}' '${ac_str_sub}'
+        substituteInPlace stalonetray.xml.in --replace '${dtd_str}' '${dtd_str_sub}'
+        autoupdate
+        aclocal
+        autoheader
+        autoconf
+        automake --add-missing
```

Comment:
I think the canonical way to get this done in the correct order is adding `autoreconfHook` to `nativeBuildInputs`.